check_puppetdb_node.rb

```diff
@@ -10,24 +10,54 @@
 require 'time'
 require 'date'
 
-def api_call(protocol="http", host="localhost", port="8080", query)
+def api_call(query, opt = {})
+  protocol     = opt.has_key?(:ssl) ? 'https' : 'http'
+  host         = opt[:host] || 'localhost'
+  port         = opt[:port] || opt.has_key?(:ssl) ? 8081 : 8080
   uri          = URI("#{protocol}://#{host}:#{port}/#{query}")
   http         = Net::HTTP.new(uri.host, uri.port)
-  http.use_ssl = true  if protocol == 'https'
-  http.use_ssl = false if protocol == 'http'
-  request      = Net::HTTP::Get.new(uri.request_uri, {'Accept' => 'application/json'})
-  response     = http.request(request)
+  http.use_ssl = opt.has_key?(:ssl)
+  if http.use_ssl?
+    require 'openssl/x509'
+    require 'openssl/pkey'
+
+    if opt.has_key?(:ca)
+      cert_store = OpenSSL::X509::Store.new
+      cert_store.add_file(opt[:ca])
+      cert_store.add_crl(OpenSSL::X509::CRL.new(File.read(opt[:crl]))) if opt.has_key?(:crl)
+      http.cert_store = cert_store
+    end
+    if opt.has_key?(:cert) and opt.has_key?(:key)
+      http.cert = OpenSSL::X509::Certificate.new(File.read(opt[:cert]))
+      http.key  = OpenSSL::PKey.read(File.read(opt[:key]))
+    end
+  end
+
+  request  = Net::HTTP::Get.new(uri.request_uri, {'Accept' => 'application/json'})
+  response = http.request(request)
   JSON.load(response.body)
 end
 
-@options = Hash.new
+@options  = Hash.new
+@http_opt = Hash.new
```

Comment:
why use a seperate options hash?

Reply:
Wanted to separate the settings so i can put the hash into the `api_call` function. Sure the function ignores not known keys but i think this makes the code more understandable, or?

Reply:
it's okay. but unusual. normally you put this all in one hash. and the functions only use the keys they access. if there are more keys in it, they don't care, because they only look for what they are told to use.

Reply:
ok merged the `@http_opt` into `@options`